src/dao/PeopleDao.cpp

```diff
@@ -298,10 +305,10 @@ bool PeopleDao::addNewTutor(Tutor *tutor) {
             return false;
         }
 
-        for (int i = 0; i < subject_count; ++i) {
+        for (int i = 0; i < subjectIds.size(); ++i) {
             db(insert_into(t_peopleSubject).set(
                     t_peopleSubject.userId = currentUserId,
-                    t_peopleSubject.subjectId = (i + 1)));
+                    t_peopleSubject.subjectId = subjectIds[i]));
```

Comment:
due to subject id is automatic increased and managed by database system , so that when some test execute will make subject ids increased by unknown order. subject id need to be confirmed by one more select operation.